scala-mode2-prettify.el

```diff
@@ -0,0 +1,54 @@
+;;; scala-mode-prettify.el - Extension of scala-mode for prettifying scala symbols/code
```

Comment:
Should the name of this file be `scala-mode-prettify-symbols.el`? As I was not familyar with the new prettify-symbols package I though this PR was about general prettifying of code (i.e. reformatting). Adding the `symbols` to the name would help others recognize instantly what this is about.